src/Behat/Testwork/ServiceContainer/Configuration/ConfigurationTree.php

```diff
@@ -30,8 +30,8 @@ final class ConfigurationTree
      */
     public function getConfigTree(array $extensions)
     {
-        $tree = new TreeBuilder();
-        $root = $tree->root('testwork');
+        $tree = new TreeBuilder('testwork');
+        $root = $tree->getRootNode();
 
```

Comment:
this won't work for 3.4 (and even less for 2.8). You should add a detection on which pattern to use (legacy or new one).

Reply:
Right, I see. I'll prepare a compatibility layer to ensure it works with all the versions.

Reply:
This is how we do it in Sylius bundles:

https://github.com/Sylius/SyliusResourceBundle/blob/a18fd0ecfa66dc5b5f76402c0ebb0c636740a54a/src/Bundle/DependencyInjection/Configuration.php#L31-L42

Reply:
Hi @pamil, I think you're approach is the best, so I'll implement the same approach